internal/controller/component_customizer.go

```diff
@@ -190,10 +190,6 @@ func customizeManagerContainer(mSpec *operatorv1.ManagerSpec, c *corev1.Containe
 		c.Args = leaderElectionArgs(mSpec.LeaderElection, c.Args)
 	}
 
-	if mSpec.Metrics.BindAddress != "" {
```

Comment:
This option is used by CAPI providers:
* CAPA: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/main.go#L452
* CAPZ: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/main.go#L591

If we delete this, users won't be able to configure metrics anymore. Plus, it's a breaking change in our API, so we have to release v1alpha3 to accept this.

Reply:
Have you checked https://github.com/kubernetes-sigs/cluster-api/pull/9264 upstream in CAPI? Both above providers are still using it, yes , but that is because they have not yet bumped to CAPI v1.6 AFAIK and 9264 suggests to remove `--metrics-bind-addr` and use `--diagnostics-address=<addr>` instead which should have same behaviour

Reply:
Interesting, but then how about other providers removing it without any conversions (I saw a couple of them: CAPO, CAPZ)?

Reply:
As this is just a configuration value, it could be left in place, with the validation webhook preventing simultaneous use of both.